Makefile

```diff
@@ -0,0 +1,17 @@
+IDIR = include

```

Comment:
Ninja seems like a great idea if we're going to emphasize on speed. The transition seems easy enough right now.
Also we had originally intended to use cmake but we hadn't gotten around to that yet so gyp it is then.